numba/core/typing/templates.py

```diff
@@ -246,7 +246,24 @@ def fold_arguments(pysig, args, kws, normal_handler, default_handler,
     return args
 
 
-class FunctionTemplate(object):
+class _Template(object):
+
+    def get_template_info(self):
+        """
+        Override to specialise the template information that will govern display
+        in error messages that appear to users"""
+        info = {
+            'kind': "unknown",
+            'name': "unknown",
+            'sig': "unknown",
+            'filename': "unknown",
+            'lines': ("unknown", "unknown"),
+            'docstring': "unknown"
+        }
+        return info
```

Comment:
This implementation seems unneeded. All subclasses are overriding this. Maybe make this a `abstractmethod` to force subclass to implement it.

Reply:
Can do, I had that at one point and then removed it as the inheritance was less clear, I think now having patched this it'd probably be fine/the right thing to do.

Reply:
3ecedbb